service/src/main/kotlin/fi/espoo/evaka/reservations/ReservationControllerCitizen.kt

```diff
@@ -32,8 +34,101 @@ class ReservationControllerCitizen {
 
         return db.read { it.getReservations(user.id, FiniteDateRange(from, to)) }
     }
+
+    @PostMapping("/citizen/reservations")
+    fun postReservations(
+        db: Database.Connection,
+        user: AuthenticatedUser,
+        @RequestBody body: ReservationRequest
+    ) {
+        Audit.AttendanceReservationCitizenCreate.log(targetId = body.children.joinToString())
+        user.requireOneOfRoles(UserRole.CITIZEN_WEAK, UserRole.END_USER)
+
```

Comment:
should we throw Forbidden already here if some child is not user's dependant? probably yes

Reply:
Maybe, if the response is the same when the child ids don't exist.